packages/devtools_app/lib/src/profiler/cpu_profiler.dart

```diff
@@ -253,3 +267,73 @@ class CpuProfilerDisabled extends StatelessWidget {
     );
   }
 }
+
+
+/// DropdownButton that controls the value of
+/// [ProfilerScreenController.userTagFilter].
+class UserTagDropdown extends StatelessWidget {
+  const UserTagDropdown(this.controller);
+
+  final CpuProfilerController controller;
+
+  @override
+  Widget build(BuildContext context) {
+    const filterByTag = 'Filter by tag:';
+    // This needs to rebuild whenever there is new CPU profile data because the
+    // user tags will change with the data.
+    return ValueListenableBuilder<CpuProfileData>(
```

Comment:
Add a todo referencing https://github.com/flutter/devtools/issues/2989
to remove the nested value listenables.

Reply:
done